src/bin/importer_offline.rs

```diff
@@ -34,13 +36,26 @@ use stratus::utils::calculate_tps_and_bpm;
 use stratus::utils::DropTimer;
 use stratus::GlobalServices;
 use stratus::GlobalState;
-use tokio::sync::mpsc;
+use tokio::sync::mpsc as async_mpsc;
 use tokio::time::Instant;
 
-/// Number of tasks in the backlog. Each task contains `--blocks-by-fetch` blocks and all receipts for them.
-const BACKLOG_SIZE: usize = 50;
+/// Number of loader tasks buffered. Each task contains `--blocks-by-fetch` blocks and all receipts for them.
+const RPC_LOADER_CHANNEL_CAPACITY: usize = 25;
 
-type BacklogTask = (Vec<ExternalBlock>, Vec<ExternalReceipt>);
+/// Size of the block batches to save after execution, and capacity of the batch channel.
+///
+/// We are using half of the inmemory temporary storage and a channel with size 1, this way, we can parallize
+/// execution and storage saving while staying below the inmemory storage limit.
+///
+/// By setting channel capacity to 1, we use backpressure to ensure that at most
+/// `INMEMORY_TEMPORARY_STORAGE_MAX_BLOCKS` are executed at a time.
+///
+/// REMOVE THIS: Decrease by one to give room for the pending block.
+/// TODO: why `max/2-1` doesn't work?
+const STORAGE_WRITER_BATCHES_SIZE: usize = INMEMORY_TEMPORARY_STORAGE_MAX_BLOCKS / 3;
```

Comment:
Auto-review requested changes: update these messages and quickly check if it's obvious why it behaves this way.